src/coreclr/vm/class.cpp

```diff
@@ -1170,7 +1170,7 @@ void ClassLoader::ValidateMethodsWithCovariantReturnTypes(MethodTable* pMT)
                 continue;
 
             Instantiation parentClassInst = pParentMD->GetClassInstantiation();
-            if (ClassLoader::IsTypicalSharedInstantiation(parentClassInst))
+            if (ClassLoader::IsCanonicalGenericInstantiation(parentClassInst))
```

Comment:
Is there a reason why this isn't if (true)?

Reply:
I am not sure I understand the question.

Reply:
Ah, you mean that for non-canonical generic instantiation in pParentMD->GetClassInstantiation(), the pParentMD->GetClassInstantiation()==pParentMT->GetInstantiation(), so we never need to check.

Reply:
@davidwrighton I've removed the `if` here and also below for the current class instantiation. All the tests are passing with that change.